mxcubeweb/__init__.py

```diff
@@ -124,9 +128,25 @@ def build_server_and_config(test=False, argv=None):
         config_path = HWR.get_hardware_repository().find_in_repository("mxcube-web")
 
         cfg = Config(config_path)
-
+        db_path = (
+            Path(cfg.flask.USER_DB_PATH)
+            if cfg.flask.USER_DB_PATH
+            else Path(FlaskConfigModel.USER_DB_PATH.default)
+        )
+        db_path.parent.mkdir(parents=True, exist_ok=True, mode=0o600)
         if test:
-            cfg.flask.USER_DB_PATH = "/tmp/mxcube-test-user.db"
+            try:
+                test_db = db_path.parent / "mxcube-test-user.db"
+                cfg.flask.USER_DB_PATH = str(test_db)
+                cfg.test_db_path = str(test_db)
+
+                # Clean up existing test database if it exists
+                if test_db.exists():
+                    test_db.unlink()
+            except AttributeError as e:
+                msg = f"Failed to set test database path: {e}"
+                logging.getLogger(msg)
+                raise
```

Comment:
This does not log anything

```suggestion
            except AttributeError as e:
                msg = "Failed to set test database path"
                logging.getLogger(SOME_LOGGER_NAME).exception(msg)
                raise
```

Reply:
Also, I have no idea what `except AttributeError` is supposed to catch.

Reply:
That was an older change that I kept and forgot to modify. We can change it to something like FileNotFoundError or/and OSError/PermissionError (subclass) in this case.

Reply:
Thanks guys. I made some simplifications after a quick chat here :)